src/components/HomePage.tsx

```diff
@@ -9,6 +11,15 @@ function HomePage() {
 
     window.location.href = `https://github.com/login/oauth/authorize?${params}`;
   };
+  const navigate = useNavigate()
+  useEffect(() => {
+    const oauthCode = typeof (localStorage?.getItem) === 'function' ? localStorage.getItem('oauthCode') : false
+    if (typeof oauthCode === 'string' && oauthCode.length > 0) {
+      const params = new URLSearchParams();
+      params.set('code', `${oauthCode}`);
+      navigate(`/oauth_redirect?${params}`)
```

Comment:
Is there a way to implement this without redirection?

Reply:
I will check